Lib/_pyio.py

```diff
@@ -198,6 +198,10 @@ def open(file, mode="r", buffering=-1, encoding=None, errors=None,
         raise ValueError("binary mode doesn't take an errors argument")
     if binary and newline is not None:
         raise ValueError("binary mode doesn't take a newline argument")
+    if binary and buffering == 1:
+        import warnings
+        warnings.warn("line buffering (buffering=1) isn't supported in binary "
+                      "mode, file will be fully buffered", RuntimeWarning, 2)
```

Comment:
There is no such thing as "fully buffered". I propose to also say "the default size will be used" here.

Extract of the code:

```
        if buffering == 1 or buffering < 0 and raw.isatty():
            buffering = -1
            line_buffering = True
        if buffering < 0:
            buffering = DEFAULT_BUFFER_SIZE
```

buffering=1 becomes buffering=DEFAULT_BUFFER_SIZE if I understand correctly.